tests/cross/src/test/scala/tests/pc/PcRenameSuite.scala

```diff
@@ -0,0 +1,332 @@
+package tests.pc
+
+import tests.BasePcRenameSuite
+
+class PcRenameSuite extends BasePcRenameSuite {
+  check(
+    "basic",
+    """|val <<a>> = 123
+       |<<@@a>> + 1  
+       |""".stripMargin,
+  )
+
+  check(
+    "generics",
+    """|trait S1[X] { def <<torename>>(p: X): String = "" }
+       |trait T1[Z] extends S1[Z] { override def <<torename>>(p: Z): String = super.<<torename>>(p) }
+       |trait T2[X] extends T1[X] { override def <<torename>>(p: X): String = super.<<torename>>(p) }
+       |trait T3[I, J] extends T2[I] { override def <<torename>>(p: I): String = super.<<torename>>(p) }
+       |trait T4[I, J] extends T3[J, I] { override def <<torename>>(p: J): String = super.<<torename>>(p) }
+       |trait T5[U] extends T4[U, U] { override def <<tore@@name>>(p: U): String = super.<<torename>>(p) }
+       |""".stripMargin,
+  )
+
+  check(
+    "match-ret-type",
+    """|trait P
+       |trait PP extends P
+       |trait A { def <<torename>>(a: String): P = ??? }
+       |trait B extends A { override def <<tore@@name>>(a: String): PP = ??? }
+       |""".stripMargin,
+  )
+
+  check(
+    "self-type",
+    """|trait <<A@@BC>>
+       |trait Alphabet{
+       |  this: <<ABC>> =>
+       |}
+       |object A{
+       |  val a = new Alphabet with <<ABC>>
+       |}
+       |""".stripMargin,
+    newName = "Animal",
+  )
+
+  check(
+    "method-inheritance",
+    """|trait Hello{
+       |  def <<method>>(abc : String) : Boolean
+       |}
+       |
+       |class GoodMorning extends Hello {
+       |  def <<met@@hod>>(abc : String) = true
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "long-inheritance",
+    """|trait A[T, S] {
+       |  def <<method>>(abc : T) : S
+       |}
+       |
+       |abstract class B[T] extends A[T, Boolean] {
+       |  def <<method>>(abc : T) : Boolean
+       |}
+       |
+       |abstract class C extends B[String] {
+       |  def <<meth@@od>>(abc : String) : Boolean = false
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "multiple-inheritance",
+    """|trait A {
+       |  def <<method>>(abc : String) : Boolean
+       |}
+       |
+       |trait B {
+       |  def <<method>>(abc : String) : Boolean = true
+       |}
+       |
+       |abstract class C extends B with A {
+       |  override def <<meth@@od>>(abc : String) : Boolean = false
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "colon-bad",
+    """|class User{
+       |  def <<:@@:>>(name : String) = name
+       |}
+       |object Main{
+       |  val user = new User()
+       |  "" <<::>> user
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "colon-good",
```

Comment:
Isn't this the same test as the previous one?

Reply:
Right, I forgot about changing `newName`